File: src/main/java/org/vors/pairbot/telegram/handler/ChatUpdateHandler.java
package org.vors.pairbot.telegram.handler;

import org.telegram.telegrambots.meta.api.objects.Update;
import org.telegram.telegrambots.meta.exceptions.TelegramApiException;

/**
 * Chain of handlers is to be executed when bot gets telegram chat update.
 */
public interface ChatUpdateHandler {
    Object handle(Update update) throws TelegramApiException;
}

File: src/main/java/org/vors/pairbot/telegram/handler/ChatUpdateHandler.kt
package org.vors.pairbot.telegram.handler

import org.telegram.telegrambots.meta.api.objects.Update
import org.telegram.telegrambots.meta.exceptions.TelegramApiException

/**
 * Chain of handlers is to be executed when bot gets telegram chat update.
 */
interface ChatUpdateHandler {
    @Throws(TelegramApiException::class)
    fun handle(update: Update): Any
}
